grafana/grafana
# Dashboard Scene: Sync variable state with TemplateSrv result

### What is this PR solving?

In the old architecture (before dashboard scene migration), plugin developers relied on `RefreshEvent`, triggered by variable changes, annotation changes, exemplars, time range changes, and other places. 

In Scenes, we are not triggering that event, because we don’t need it. However, this is considered a breaking change for 11.3.0, and we needed to trigger that `Refresh Event` in the scenes migration workflow 

The issue was first reported at: https://github.com/VolkovLabs/business-variable/issues/192 

### Changes:
- [x] Verify Query variable options are populated in the plugin
- [x] Add refresh event when variable changes
- [x] Add refresh event when annotation changes
- [x] Add unit test in Dashboard Scene
- [ ] Check exemplar's behavior 

### How to test this PR:
-  In your local Grafana instance install the panel plugin http://localhost:3000/plugins/volkovlabs-variable-panel 
- Create a dashboard and create a new panel with the viz  `Business Variable`
- Create a custom variable with a multi-select set
- Edit the panel and select the newly custom variable
- Verify the variables are updated in the panel and the dropdown 

https://github.com/user-attachments/assets/ce199e7e-4201-47dc-8c3a-63145f4ac927
